### DISCUSSION TITLE
Custom Rule Based Translator

### DISCUSSION THREAD

**Discussion by iuithor** [sign-language-translator/sign-language-translator#43] [Q&A]
Hello! 👋 
I am trying to build a model to translate from text to (Chilean) sign language. I'm having some trouble understanding the reading documents, mostly because I'm relatively new working with AI and its training.

My question is, is there any more detailed / friendly guide? Or any community where they deal with this specific package? 

Many thanks :-)

**Comment by mdsrqbl** [comment]
Hi,

The rule-based model does not require any expertise in AI. It will work for chilean language if you override the `slt.lanngauges.TextLanguage` & `slt.languages.SignLanguage` classes. You may see the code for the languages I have implemented already but it will most likely change in the upcoming versions.

The readmes, [notebooks](https://github.com/sign-language-translator/notebooks/tree/main/translation) and [readTheDocs](https://slt.readthedocs.io/) is all I've made so far. The documentation is often outdated and I need the user's perspective to write the best walkthrough. 

Feel free to ask the specifics and I'll guide you & update the docs accordingly.

**Reply by iuithor** [reply]
My problem is with the classes, when I write "pass", it says that it needs the parameters to be able to execute it, but I'm not finding/understanding what exactly I have to write. 

The guides, I think, are fine! I'm just a little clumsy and need clear examples or I end up getting confused fast.
